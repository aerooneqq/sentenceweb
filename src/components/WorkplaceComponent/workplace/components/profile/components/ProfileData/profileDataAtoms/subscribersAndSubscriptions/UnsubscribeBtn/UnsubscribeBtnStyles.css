.unsubscribeBtn { 
    width: 80px;
    height: 30px;

    background: orange;

    border-style: solid;
    border-width: 1px;
    border-radius: 7px;
    border-color: orange;

    font-family: Arial;
    color: white;
}

.unsubscribeBtn:focus{ 
    outline: none;
}

.unsubscribeBtn:hover { 
    background: linear-gradient(-120deg, darkorange, orange)
}