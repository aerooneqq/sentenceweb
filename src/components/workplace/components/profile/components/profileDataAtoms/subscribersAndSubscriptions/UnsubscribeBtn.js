import React, {Component} from "react";

//Styles
import "./styles/UnsubscribeBtnStyles.css";

export default class UnsubscribeBtn extends Component{ 
    constructor(props){ 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick(){ 
        this.props.deleteSubscription(this.props.userID);
    }

    render() { 
        return (
            <button className = "unsubscribeBtn"
                    onClick = {this.handleBtnClick}>
                Unsub
            </button>
        )
    }
}