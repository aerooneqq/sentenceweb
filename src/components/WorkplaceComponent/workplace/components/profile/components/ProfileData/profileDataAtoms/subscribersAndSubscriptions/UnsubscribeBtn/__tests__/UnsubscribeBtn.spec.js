import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import UnsubscribeBtn from "../UnsubscribeBtn";

const propsSpy = jest.fn(() => {});
const props = { 
    userID: 1,
    deleteSubscription: propsSpy
}

describe("UnsubscribeBtn", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<UnsubscribeBtn />))).toMatchSnapshot();
    });

    it("should call deleteSubscriptions from props", () => { 
        const unsubscribeBtn = shallow(<UnsubscribeBtn {...props}/>);

        unsubscribeBtn.find("button").simulate("click");

        expect(propsSpy).toBeCalled();
    });

    it("should call handleBtnClick when clicked", () => { 
        const spy = jest.spyOn(UnsubscribeBtn.prototype, "handleBtnClick").mockImplementation(() => {});
        const unsubscribeBtn = shallow(<UnsubscribeBtn {...props}/>);

        unsubscribeBtn.find("button").simulate("click");

        expect(spy).toBeCalled();
    });
}); 