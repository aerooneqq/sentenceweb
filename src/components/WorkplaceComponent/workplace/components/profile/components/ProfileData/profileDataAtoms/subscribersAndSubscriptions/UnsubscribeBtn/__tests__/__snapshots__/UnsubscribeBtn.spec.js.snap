// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsubscribeBtn should render correctly 1`] = `
<button
  className="unsubscribeBtn"
  onClick={[Function]}
>
  Unsub
</button>
`;
